### DISCUSSION TITLE
Nuxt content config markdown anchor links override custom ProseA component

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/content#2281] [Q&A]
I created a custom `ProseA.vue` component that I have in the `/components/content` directory and it works just fine, but when I set `content.markdown.achorLinks = true` in `nuxt.config.ts`, it overrides my custom ProseA component. I have the below config and I thought I could prevent this by utilizing the tags option but this does not seem to work either

```js
export default defineNuxtConfig({
  content: {
    markdown: {
      anchorLinks: true,
      tags: {
        a: 'ProseA'
      }
    }
  }
})
```

**Comment by nobkd** [comment]
First of all. Your `a: ProseA` does nothing, as the default component for `a` in Markdown is already `ProseA`. It is just not used by ProseH1 - H6.

At the start thought of something else, so the content of the parantheses can also be skipped, but might be useful to understand, why using ProseA from within ProseH1-H6 might be a bad idea ( as it also leverages NuxtLink (which is good)):

(
For the link generation on titles there was used normal html `<a>` tag, because `<NuxtLink>` didn't properly work with anchor links.

See #1380 with #1381

This might have been fixed, but I haven't checked that for some time.
) 

Also, there was never `<ProseA>` involved, as it only is used from within Markdown.

To solve the problem on your side, you can override h1-h6 and add the styles to the links there.
You can find the latest components here: https://github.com/nuxt-modules/mdc/tree/main/src%2Fruntime%2Fcomponents%2Fprose

**Reply by dbzx10299** [reply]
I have a custom `<ProseA />` component and in my markdown I have to write
```
## [Level Two](#some-fragment)
```
This works well with `markdown.anchorLinks = 'false'` however I have a ton of headings I need to convert to this syntax and I have a certain style that needs to be kept due to a custom intersection observer component. 

Thats why the links need to be in a certain format and nuxt content doesn‘t use my ProseA component. 

It looks like I will have to manually convert all my headings but I wasn‘t sure if there was a way around this. When `markdown.anchorLinks = true` there is indeed and anchor tag but It doesn‘t seem to be used as ProseA

**Reply by nobkd** [reply]
As mentioned, when headings create anchor links with anchorLinks option, a default html `<a>` is used.
ProseA isn't used by heading generation, so of course the styles aren't applied.
If you want to use ProseA for the heading generation, you can override the components for H1 up to H6 and replace the `a` tags with `ProseA` ones.

A possible problem by doing so might be, that scrolling to the headings might not work properly with NuxtLink which is used by ProseA. But I'm not sure about this anymore, because much code changed.
